### DISCUSSION TITLE
Redirect when not authenticated

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#740] [Q&A] [answered]
Currently, the `current_user()` dependency callable [throws an exception when the user is not authenticated](https://github.com/fastapi-users/fastapi-users/blob/eb47dc8e752964a72ffa61333179a811771318ba/fastapi_users/authentication/__init__.py#L161). I was wondering how we can change this behavior if e.g. we want to redirect to a login page. It is possible to [redirect from a dependency](https://github.com/tiangolo/fastapi/issues/1039#issuecomment-591661667), but it's a bit complicated as it requires registering an exception handler on the FastAPI app.
Maybe this could be handled by FastAPI-Users, with something like `current_user(redirect="my_url")`, or by registering the redirect URL in the user manager?

**Comment by frankie567** [comment, marked as answer]
I see the point here. The thing is, at its root, FastAPI (and FastAPI Users) are mainly designed for REST API. That's why we return hard 401 error when not authenticated.

To implement it in FastAPI Users, we would need to change [that part](https://github.com/fastapi-users/fastapi-users/blob/0826ec529915e80c217f1ddc52f76d1c1c71d223/fastapi_users/authentication/__init__.py#L127-L162) and allow to customize the `HTTPException` attributes. The status code would have to be 302 with a `Location` header pointing to the login page. That sounds a bit counter-intuitive to me 🤔

What I do personally is to handle this on the frontend side: if I can't retrieve the currently logged in user because of a 401, I can trigger the redirection to the login page.
